src/main/java/games/strategy/triplea/delegate/NonFightingBattle.java

```diff
@@ -32,10 +33,19 @@
  */
 public class NonFightingBattle extends DependentBattle {
   private static final long serialVersionUID = -1699534010648145123L;
+  private Set<Territory> m_attackingFrom;
+  private Collection<Territory> m_amphibiousAttackFrom;
+  private Map<Territory, Collection<Unit>> m_attackingFromMap;
 
+  /**
+   *  Constructor. Suppress checkstyle warning.
+   */
   public NonFightingBattle(final Territory battleSite, final PlayerID attacker, final BattleTracker battleTracker,
       final GameData data) {
-    super(battleSite, attacker, battleTracker, false, BattleType.NORMAL, data);
+    super(battleSite, attacker, battleTracker, data);
+    m_attackingFromMap = new HashMap<>();
+    m_attackingFrom = new HashSet<>();
+    m_amphibiousAttackFrom = new ArrayList<>();
```

Comment:
Can these initializations be moved to the field declaration?

Also, IIRC, all of these fields were `final` prior to #1583.  Can they be made so again?